models/user_model.go

```diff
@@ -12,10 +12,13 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-func FindProfile(username string) (result map[string]interface{}, err error) {
+func FindProfile(username string, userType string) (result map[string]interface{}, err error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	filter := bson.D{{Key: "username", Value: username}}
+	if userType != "" {
+		filter = append(filter, bson.E{Key: "usertype", Value: userType})
+	}
 	opts := options.Find().SetProjection(bson.D{{"_id", 0}, {"hashedPassword", 0}, {"createdAt", 0}, {"updatedAt", 0}})
```

Comment:
This part looks kinda weird to me. It seems that there is a FindOne function that you could use can could reduce a lot of code in this part. Please look in to that if it is not possible to use FindOne, then reply to this comment.